### PR TITLE
Adjustments for XAML generation info

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

- Use `CreateNewOnMetadataUpdate` for XAML generated types.
- Remove unstable generation info processing.

### <samp>🤖 Generated by Copilot at 02c3c8a</samp>

Simplified the XAML code generation logic and added support for hot reload with metadata changes. Removed unused code and added a new `CreateNewOnMetadataUpdate` attribute to mark types for recreation.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 1]
```diff
@@ -361,42 +354,16 @@ private SourceText InnerGenerateFile()
 			{
 				_isTopLevelDictionary = true;
 
-				if (_generationRunFileInfo.RunInfo.Manager.IsFirstRun(_generationRunFileInfo) || !_useXamlReaderHotReload)
-				{
-					// On the first run, or if XamlReader hot reload is disabled, generate the full code.
-
-					var componentBuilder = new IndentedStringBuilder();
+				var componentBuilder = new IndentedStringBuilder();
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:357]: Consider removing `componentBuilder` completely and just appending to `writer` directly.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 2]
```diff
@@ -409,63 +376,41 @@ private SourceText InnerGenerateFile()
...
 					using (writer.BlockInvariant("partial class {0} : {1}", _xClassName.ClassName, controlBaseType.GetFullyQualifiedTypeIncludingGlobal()))
 					{
 						BuildBaseUri(writer);
 
 						using (Scope(_xClassName.Namespace, _xClassName.ClassName))
 						{
-							if (_generationRunFileInfo.RunInfo.Manager.IsFirstRun(_generationRunFileInfo) || !_useXamlReaderHotReload)
-							{
-								var componentBuilder = new IndentedStringBuilder();
+							var componentBuilder = new IndentedStringBuilder();
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:387]: Same here

**Review by Youssef1313 — APPROVED** [review]
And finally, we are removing this code! 🎉

**Review by Youssef1313 — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 3]
```diff
@@ -361,42 +354,16 @@ private SourceText InnerGenerateFile()
...
-					using (componentBuilder.Indent(writer.CurrentLevel))
-					{
-						BuildResourceDictionaryBackingClass(componentBuilder, topLevelControl);
-						BuildTopLevelResourceDictionary(componentBuilder, topLevelControl);
-					}
-
-					var componentCode = componentBuilder.ToString();
-					if (_useXamlReaderHotReload)
-					{
-						_generationRunFileInfo.SetAppliedTypes(_xamlAppliedTypes);
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:379]: `_xamlAppliedTypes` isn't used now I think. The field can be removed

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 4]
```diff
@@ -409,63 +376,41 @@ private SourceText InnerGenerateFile()
...
+								BuildBackingFields(componentBuilder);
 
-									BuildXBindTryGetDeclarations(componentBuilder);
-								}
+								BuildChildSubclasses(componentBuilder);
 
-								var componentCode = componentBuilder.ToString();
+								BuildComponentFields(componentBuilder);
 
-								if (_useXamlReaderHotReload)
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:447]: `_useXamlReaderHotReload` is no longer used I think?

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs (hunk 3 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:379]: it's still used, it's present to optimize `Apply()` invocations

**Pull request merged**